common/src/main/java/dev/isxander/yacl3/gui/controllers/ColorPickerElement.java

```diff
@@ -0,0 +1,436 @@
+            mouseDown = true;
+            hueSliderDown = false;
+            satLightGradientDown = false;
+            alphaSliderDown = false;
+            setColorFromMouseClick(mouseX, mouseY);
+            return true;
+        } else if(entryWidget.isMouseOver(mouseX, mouseY)) {
+            return entryWidget.mouseClicked(mouseX, mouseY, button);
+        } else {
+            if(Minecraft.getInstance().screen instanceof PopupControllerScreen popupControllerScreen) {
+                popupControllerScreen.onClose();
+            }
+            return false;
+        }
+    }
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        mouseDown = false;
+        return false;
+    }
+    @Override
+    public boolean isMouseOver(double mouseX, double mouseY) {
+        //Checks if the mouse is either over the color picker or the color controller
+        //The addition/subtraction of the outline and extra 3 pixels is to account for both the outline and the background
+        if (mouseX >= colorPickerDim.x() - outline - 3 && mouseX <= colorPickerDim.xLimit() + outline + 3
+                && mouseY >= colorPickerDim.y() - outline - 3 && mouseY <= colorPickerDim.yLimit() + outline + 3) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean mouseDragged(double mouseX, double mouseY, int button, double deltaX, double deltaY) {
+        if(mouseDown || isMouseOver(mouseX, mouseY)) {
+            setColorFromMouseClick(mouseX, mouseY);
+            return true;
+        }
+        return entryWidget.mouseDragged(mouseX, mouseY, button, deltaX, deltaY);
+    }
+
+    @Override
+    public boolean charTyped(char chr, int modifiers) {
+        //Done to allow for typing whilst the color picker is visible
+        charTyped = true;
+        return entryWidget.charTyped(chr, modifiers);
+    }
+
+    @Override
+    public void setDimension(Dimension<Integer> dim) {
+        super.setDimension(dim);
+
+        int colorPickerHeight = (dim.height() * 2) + 7;
+        int colorPickerX = dim.centerX() - getXPadding() * 2;
+        int colorPickerY = dim.y() - colorPickerHeight - sliderHeight;
+        int alphaSliderHeight = 0;
+        if(controller.allowAlpha()) {
+            alphaSliderHeight = sliderHeight + outline + paddingY;
+            colorPickerHeight += alphaSliderHeight;
+            colorPickerY -= alphaSliderHeight;
+        }
+
+        //A single dimension for the entire color picker as a whole
+        //Division is used for the main color preview, saturation/light picker, and hue slider to determine their dimensions
+        colorPickerDim = Dimension.ofInt(colorPickerX, colorPickerY, dim.xLimit() - colorPickerX, colorPickerHeight);
+
+        //Add anything after the colorPickerDim.height to reduce the "strictness" of the detection
+        if(colorPickerDim.yLimit() - colorPickerDim.height() < screen.tabArea.top()){
```

Comment:
I'm trying to understand this. Is this the same as `if (colorPickerY < screen.tabArea.top())` or am I missing something?

Reply:
I think so, yeah. That's my bad.